Polish principles sections (#149)
myyoda/principles-paper: main.tex

\end{itemize}

Reproducibility fails when a research object depends on resources that are not part of it; an undocumented library, a dataset referenced by a broken URL, a script that assumes tools are installed on the host system.

> We call this the ``don't look up'' rule: a research object must never rely on an implicit external state.

Reviewer: I do not like this, IMO its good as is. But if you prefer I'd be also ok with:
```suggestion
We call this the ``don't look up'' rule: a research object must never rely on any implicit external state.
```

    all components involved.
\end{itemize}

> Research data and code change constantly during a project, and each change compounds the difficulty of reconstructing any previous state, such as which version of a dataset was used for a particular analysis, what parameters were set, what code produced a given figure.

Reviewer: ```suggestion
Research data and code change constantly during a project, and each change compounds the difficulty of reconstructing any previous state, such as which version of a dataset was used for a particular analysis, what parameters were set, and what code produced a given figure.
```

This distinction matters; two datasets labeled ``version 1.0'' by different labs are ambiguous whereas two datasets with identical content hashes are provably identical.
Version control commits also capture basic provenance of who made the change, when, and a description of why.
This unifies identity and provenance in a single system and also provides a safety net since any change can be reverted and any prior state restored, making it easier to modify and experiment.
For tracking to be effective, all components of the research object must be under version control: code, data, environment definitions, and configuration.

Traditional version control is designed for code and other text files, but research objects often include large binary datasets, container images, and other assets that do not fit this model.
Compatible tools such as git-annex, DataLad, or DVC extend content-addressed tracking to these components, ensuring that all parts of the research object are managed under the same system.

Toward the ideal end, provenance is captured programmatically by tooling rather than by manual annotation, producing richer records than a human would typically write.

> Each computational step then records exactly what command was executed and what inputs it consumed.

Reviewer: IMO this flows a little better wdyt?

```suggestion
Toward the ideal end, provenance is captured programmatically by tooling rather than by manual annotation.
Each computational step should record exactly what command was executed and what inputs were consumed, producing richer and more reliable records than a human would typically write.
```

\anchoredsubsection{portability}{Portability}

\begin{itemize}

>     \item \textbf{P.1}: Procedures MUST NOT depend on an undocumented host environment state.

Reviewer: Same as above. Prefer as is, this is also acceptable:
```suggestion
    \item \textbf{P.1}: Procedures MUST NOT depend on any undocumented host environment state.
```

Portability requires making all host dependencies explicit, so that the research object can be adapted to a new environment.

Some degree of coupling to the host is unavoidable, such as path naming restrictions (``C:/'' vs. ``/mnt''), resource limits (minimum CPU/RAM requirements), or platform-specific flags.
The spectrum of Portability begins with isolating this host-specific configuration from the rest of the research object, so that adapting to a new environment, however involved that may be, requires only changing the configuration.
Beyond this, several complementary approaches reduce host coupling.

> Virtual machines provide full system-level isolation; container-based tools (Docker, Singularity/Apptainer) bundle OS-level environments; declarative package managers (Nix, Guix) specify environments that can be reconstructed from a definition.

Author: The semi-colon chain can be nice for breaking up our lists of independent ideas into digestible pieces

I've started to use it consistently here
